pyvista/core/filters/data_set.py

```diff
@@ -7105,10 +7109,20 @@ def transform(  # type: ignore[misc]
         ...         [0, 0, 0, 1],
         ...     ]
         ... )
-        >>> transformed = mesh.transform(transform_matrix)
+        >>> transformed = mesh.transform(transform_matrix, inplace=False)
         >>> transformed.plot(show_edges=True)
 
         """
+        if inplace is None:
+            warnings.warn(
+                """
+The default value of `inplace` for the `dataset.transform` filter will change in the future.
+ Previously it defaulted to `True`, but will change to `False`.
+ Explictly set `inplace` to `True` or `False` to silence this warning. """,
+                PyVistaDeprecationWarning,
```

Comment:
Is a `PyVistaFutureWarning` more appropriate? 

```suggestion
                PyVistaFutureWarning,
```

Though I don't see reference to this warning in CONTRIBUTING, not sure what the policy should be here. #1860 also used `PyVistaDeprecationWarning` for the same purpose, though it's also an older PR...



A future warning is used by `extrude` though for the same kind of default value change:
https://github.com/pyvista/pyvista/blob/f9fc22d304b37a0e8f91207dee55b0c6ee75f92f/pyvista/core/filters/poly_data.py#L3086-L3090

Reply:
Yeah I always lean pretty heavily on CONTRIBUTING for this kinda stuff because I can never remember what's exactly expected. If PyVistaFutureWarning is preferred would like CONTRIBUTING updated / clarified

Reply:
Let's keep deprecation then.